modules/video/src/tracking/tracker_nano.cpp

```diff
@@ -0,0 +1,360 @@
+
+    std::vector<float> x1Vec(sz, 0);
+
+    for (int i = 0; i < sz; i++)
+    {
+        x1Vec[i] = i - sz2;
+    }
+
+    Mat x1M(1, sz, CV_32FC1, x1Vec.data());
+
+    cv::repeat(x1M, sz, 1, grid2searchX);
+    cv::repeat(x1M.t(), 1, sz, grid2searchY);
+
+    grid2searchX *= trackState.totalStride;
+    grid2searchY *= trackState.totalStride;
+
+    grid2searchX += trackState.instanceSize/2;
+    grid2searchY += trackState.instanceSize/2;
+}
+
+void TrackerNanoImpl::init(InputArray image_, const Rect &boundingBox_)
+{
+    scoreSize = (trackState.instanceSize - trackState.exemplarSize) / trackState.totalStride + 8;
+    trackState = trackerConfig();
+    image = image_.getMat().clone();
+
+    // convert Rect2d from left-up to center.
+    targetPos[0] = float(boundingBox_.x) + float(boundingBox_.width) * 0.5f;
+    targetPos[1] = float(boundingBox_.y) + float(boundingBox_.height) * 0.5f;
+
+    targetSz[0] = float(boundingBox_.width);
+    targetSz[1] = float(boundingBox_.height);
+
+    imgSize = image.size();
+
+    // Extent the bounding box.
+    float sumSz = targetSz[0] + targetSz[1];
+    float wExtent = targetSz[0] + trackState.contextAmount * (sumSz);
+    float hExtent = targetSz[1] + trackState.contextAmount * (sumSz);
+    int sz = int(cv::sqrt(wExtent * hExtent));
+
+    Mat crop;
+    getSubwindow(crop, image, sz, trackState.exemplarSize);
+    Mat blob = dnn::blobFromImage(crop, 1.0, Size(), Scalar(), trackState.swapRB);
+
+    backbone.setInput(blob);
+    Mat out = backbone.forward(); // Feature extraction.
+    neckhead.setInput(out, "input1");
+
+    createHanningWindow(hanningWindow, Size(scoreSize, scoreSize), CV_32F);
+    generateGrids();
+}
+
+void TrackerNanoImpl::getSubwindow(Mat& dstCrop, Mat& srcImg, int originalSz, int resizeSz)
+{
+    Scalar avgChans = mean(srcImg);
+    Size imgSz = srcImg.size();
+    int c = (originalSz + 1) / 2;
+
+    int context_xmin = targetPos[0] - c;
+    int context_xmax = context_xmin + originalSz - 1;
+    int context_ymin = targetPos[1] - c;
+    int context_ymax = context_ymin + originalSz - 1;
+
+    int left_pad = std::max(0, -context_xmin);
+    int top_pad = std::max(0, -context_ymin);
+    int right_pad = std::max(0, context_xmax - imgSz.width + 1);
+    int bottom_pad = std::max(0, context_ymax - imgSz.height + 1);
+
+    context_xmin += left_pad;
+    context_xmax += left_pad;
+    context_ymin += top_pad;
+    context_ymax += top_pad;
+
+    Mat cropImg;
+    if (left_pad ==0 && top_pad == 0 && right_pad == 0 && bottom_pad == 0)
+    {
+        // Crop image without padding.
+        cropImg = srcImg(cv::Rect(context_xmin, context_ymin,
+                                  context_xmax - context_xmin + 1, context_ymax - context_ymin + 1));
+    }
+    else // Crop image with padding, and the padding value is avgChans
+    {
+        cv::Mat tmpMat = cv::Mat::zeros(imgSz.height + top_pad + bottom_pad, imgSz.width + left_pad + right_pad, srcImg.type());
```

Comment:
cv::Mat::zeros() is redundant. `cv::copyMakeBorder` Fills the same area with new values.